rclpy/src/rclpy/node.cpp

```diff
@@ -441,7 +443,22 @@ Node::Node(
     new rcl_node_t,
     [](rcl_node_t * node)
     {
-      rcl_ret_t ret = rcl_node_fini(node);
+      rcl_ret_t ret;
+      {
+        rclpy::LoggingGuard scoped_logging_guard;
+        if (rcl_logging_rosout_enabled()) {
```

Comment:
I think we need to pass the enable_rosout parameter into lambda and add the check.

Reply:
see https://github.com/ros2/rclcpp/pull/2174/files#r1332059703

Reply:
okay my expectation does not work with rclpy, this fails on https://github.com/ros2/rclpy/blob/0a0239598a635f59d934851e2af3479fabc1c5da/rclpy/test/test_create_node.py#L61.

since rclpy does call `rcl_node_init` after this shared pointer is created, that said if `rcl_node_init` fails in the way, this generates the segmentation fault.

Reply:
capture `enable_rosout` argument flag. addressed in https://github.com/ros2/rclpy/commit/5e97177e97a2491e722b9543b276da0587052d94